azure-devops/azext_devops/dev/pipelines/pipeline_folders.py

```diff
@@ -0,0 +1,101 @@
+    :param detect: Automatically detect organization and project. Default is "on".
+    :type detect: str
+    """
+    organization, project = resolve_instance_and_project(
+        detect=detect, organization=organization, project=project)
+    client = get_build_client(organization)
+    from azext_devops.devops_sdk.v5_0.build.models import Folder
+    folder = Folder()
+    if description:
+        folder.description = description
+    folder.path = path
+    new_folder = client.create_folder(folder=folder, path=path, project=project)
+    return new_folder
+
+
+def pipeline_folder_delete(path, organization=None, project=None, detect=None):
+    """ Delete a folder.
+    :param path: Full path of the folder.
+    :type path: str
+    :param project: Name or ID of the team project.
+    :type project: str
+    :param detect: Automatically detect organization and project. Default is "on".
+    :type detect: str
+    """
+    organization, project = resolve_instance_and_project(
+        detect=detect, organization=organization, project=project)
+    client = get_build_client(organization)
+    return client.delete_folder(path=path, project=project)
+
+
+def pipeline_folder_list(path=None, query_order=None, organization=None, project=None, detect=None):
+    """ List all folders.
+    :param path: Full path of the folder.
+    :type path: str
+    :param query_order: Order in which folders are returned.
+    :type query_order: str
+    :param project: Name or ID of the team project.
+    :type project: str
+    :param detect: Automatically detect organization and project. Default is "on".
+    :type detect: str
+    """
+    organization, project = resolve_instance_and_project(
+        detect=detect, organization=organization, project=project)
+    client = get_build_client(organization)
+    if query_order:
+        if query_order.lower() == 'asc':
+            query_order = 'folderAscending'
+        elif query_order.lower() == 'desc':
+            query_order = 'folderDescending'
+    return client.get_folders(path=path, query_order=query_order, project=project)
+
+
+def pipeline_folder_update(path, new_path=None, new_description=None,
+                           organization=None, project=None, detect=None):
+    """ Update a folder name or description.
+    :param path: Full path of the folder.
+    :type path: str
+    :param new_path: New full path of the folder.
+    :type new_path: str
+    :param new_description: New description of the folder.
+    :type new_description: str
+    :param project: Name or ID of the team project.
+    :type project: str
+    :param detect: Automatically detect organization and project. Default is "on".
+    :type detect: str
+    """
+    organization, project = resolve_instance_and_project(
+        detect=detect, organization=organization, project=project)
+    client = get_build_client(organization)
+    folders = client.get_folders(path=path, project=project)
+    if len(folders) > 1:
```

Comment:
just curious, when this can happen?

Reply:
This is because the List API documentation says -
![image](https://user-images.githubusercontent.com/11888714/64757086-97a74d00-d54e-11e9-9e52-ffd77b834caf.png)
but it is not working in that way so right now I am only getting one path which is the exact match.

Just to handle it better now I am checking against the list to find a matching path and update the matching path. Only error out if the matching path is not found.